server/routerlicious/packages/kafka-orderer/src/kafkaOrderer.ts

```diff
@@ -248,7 +248,17 @@ export class KafkaOrdererFactory {
 		return orderer;
 	}
 
-	public delete(tenantId: string, documentId: string): void {
-		this.ordererMap.delete(`${tenantId}/${documentId}`);
+	public async delete(tenantId: string, documentId: string): Promise<void> {
+		const fullId = this.getOrdererMapKey(tenantId, documentId);
+
+		const orderer = this.ordererMap.get(fullId);
+		if (orderer !== undefined) {
+			this.ordererMap.delete(fullId);
+			await (await orderer).close();
```

Comment:
Maybe overthinking it, but could there be some concern with awaiting on the object we deleted from ordererMap. Could it somehow get garbage collected before execution returns to the await?

Reply:
Hmm I don't think so because `orderer` is being referenced by this function's stack frame even after the reference is deleted from the `ordererMap`. This stack frame won't complete until `orderer.close()` completes, so it can't be garbage collected before then.